packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -0,0 +1,1099 @@
+      center: childRect.center,
+      width: childRect.width * _kOpenScale,
+      height: childRect.height * _kOpenScale,
+    );
+
+    // TODO(justinmc): Using Overlay.of(context) means that this pops on top of
+    // the AppBar, but it doesn't on native. Is there any way around this? The
+    // reason that I do this in an Overlay is that it needs to sit on top of
+    // widgets adjacent to it.
+    // Also, I noticed that I'm having the same theme problem where a widget in
+    // the Overlay might not be themed the same as it is otherwise. Sizing can
+    // also potentially be a problem if the child was sized by its parent before.
+    return OverlayEntry(
+      opaque: false,
+      builder: (BuildContext context) {
+        return _DecoyChild(
+          beginRect: childRect,
+          child: widget.child,
+          controller: _decoyController,
+          endRect: _decoyChildEndRect,
+        );
+      },
+    );
+  }
+
+  void _onDecoyAnimationStatusChange(AnimationStatus animationStatus) {
+    switch (animationStatus) {
+      case AnimationStatus.dismissed:
+        if (_route == null) {
+          setState(() {
+            _childOpacity = 1.0;
+          });
+        }
+        _lastOverlayEntry?.remove();
+        _lastOverlayEntry = null;
+        break;
+
+      case AnimationStatus.completed:
+        setState(() {
+          _childOpacity = 0.0;
+        });
+        _openContextMenu();
+        // TODO(justinmc): I have to keep this on the screen for 1 frame because
+        // otherwise I see a white flash. I think it's because I have to render
+        // one frame offscreen. Any better way to do this?
+        Future<void>.delayed(const Duration(milliseconds: 1)).then((_) {
+          _lastOverlayEntry?.remove();
+          _lastOverlayEntry = null;
+          _decoyController.reset();
+        });
+        break;
+
+      default:
+        return;
+    }
+  }
+
+  // Watch for when _ContextMenuRoute is closed.
+  void _routeAnimationStatusListener(AnimationStatus status) {
+    if (status != AnimationStatus.dismissed) {
+      return;
+    }
+    setState(() {
+      _childOpacity = 1.0;
+    });
+    _route.animation.removeStatusListener(_routeAnimationStatusListener);
+    _route = null;
+  }
+
+  void _onTap() {
+    if (_decoyController.isAnimating) {
+      _decoyController.reverse();
+    }
+  }
+
+  void _onTapCancel() {
+    if (_decoyController.isAnimating) {
+      _decoyController.reverse();
+    }
+  }
+
+  void _onTapUp(TapUpDetails details) {
+    if (_decoyController.isAnimating) {
+      _decoyController.reverse();
+    }
+  }
+
+  void _onTapDown(TapDownDetails details) {
+    setState(() {
+      _childOpacity = 0.0;
+    });
+    _lastOverlayEntry = _overlayEntry;
+    Overlay.of(context).insert(_lastOverlayEntry);
+    _decoyController.forward();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTapCancel: _onTapCancel,
```

Comment:
Is it possible to use a `LongPressGestureRecognizer`, instead of using `isAnimating` to tell if the gesture should be accepted as a long press?

Reply:
Compromise: I changed this to base the whole animation on the duration of a long press, which I think is a lot more robust than my original approach.  I didn't directly use `onLongPress` though.